pinot-core/src/main/java/com/linkedin/pinot/core/operator/MCombineGroupByOperator.java

```diff
@@ -172,9 +171,8 @@ private IntermediateResultsBlock combineBlocks()
         public void runJob() {
           AggregationGroupByResult groupByResult;
 
-          // If block is null (unexpected error), or there's no group-by result, then return as there's nothing to do.
-          blocks[index] = (IntermediateResultsBlock) _operators.get(index).nextBlock();
-          if (blocks[index] != null) {
+          try {
+            blocks[index] = (IntermediateResultsBlock) _operators.get(index).nextBlock();
```

Comment:
Yes, that's true, and the null pointer exception will be caught in the catch block. (Actually when any exception happened, we are already out of this block.)